### DISCUSSION TITLE
How to display all the available results to a search query in HTML

### DISCUSSION THREAD

**Discussion by micaelagimenez** [community/community#21478] [Other Feature Feedback, Questions, & Ideas]
<p>I have this search engine in django that when you look for let’s say “release date” only displays one result with that date, but not all the available results with it. I was wondering how I can manage that, so that if I look for something that has more than one result, it will show all of them and not a random unique result.</p>
<p>This is the search function in my view:</p>
<pre><code class="lang-auto">def films(request):
   movies = []
   
   if request.method == 'POST':
       
       film_url = 'https://ghibliapi.herokuapp.com/films/'
       
       search_params = {
           'films' : 'title',
           'films' : 'description',
           'films' : 'director',
           'films' : 'release_date',
           'q' : request.POST['search']
           
           }

       
       r = requests.get(film_url, params=search_params)
       results = r.json()
       
       if len(results):
           for result in results:
               movie_data = {
               'Title' : result['title'],
               'Release_date': result['release_date'],
               'Director' : result['director'],
               'Producer' : result['producer'],
               'Description' : result['description']

           }

           movies.append(movie_data)  
       
    
       else:
           message = print('No results found')

   context = {
   'movies' : movies
   }
       

   return render(request,'core/films.html', context)
</code></pre>
<p>This is my html:</p>
<pre><code class="lang-auto">{% load static %}
&lt;!DOCTYPE html&gt;
&lt;html&gt;
&lt;head&gt;
   &lt;title&gt;Ghibli Studio | Movies&lt;/title&gt;
   &lt;link rel="stylesheet" href="{% static 'core/films.css' %}"&gt;
&lt;/head&gt;
&lt;body&gt;
   &lt;div class="layer"&gt;
       &lt;div class=" header"&gt;

       &lt;/div&gt;
       &lt;div class="wrap"&gt;
           &lt;form action='/films' method="POST"&gt;
               {% csrf_token %}
               &lt;div class="search"&gt;

                   &lt;input type="text" name="search" class="searchTerm" placeholder="     Type movie name"&gt;
                   &lt;link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"&gt;
                   &lt;button type="submit" class="searchButton"&gt;

                       &lt;i class="fa fa-search" style="font-size:24px"&gt;&lt;/i&gt;
                   &lt;/button&gt;

               &lt;/div&gt;

           &lt;/form&gt;
       &lt;/div&gt;

       {% if movies %}
       {% for movie in movies %}
       &lt;div&gt;
           &lt;ul class="result"&gt;
               &lt;style&gt;
                   @import url('https://fonts.googleapis.com/css2?family=Dosis:wght@300&amp;display=swap');
               &lt;/style&gt;

               &lt;h4&gt;{{movie.Title}}&lt;/h4&gt;
               &lt;h5 style="color:lightslategray;"&gt;
                   Release date: {{movie.Release_date}} &lt;br&gt;
                   Director: {{movie.Director}}
                   &lt;br&gt;Producer: {{movie.Producer}}
                   &lt;br&gt;
                   &lt;br&gt;{{movie.Description}}
               &lt;/h5&gt;

           &lt;/ul&gt;
           {% endfor %}

           {% endif %}
       &lt;/div&gt;
       
   &lt;/div&gt;
&lt;/body&gt;
&lt;/html&gt;
</code></pre>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="128776">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<pre><code class="lang-auto">           for result in results:
               movie_data = {
               'Title' : result['title'],
               'Release_date': result['release_date'],
               'Director' : result['director'],
               'Producer' : result['producer'],
               'Description' : result['description']

           }

           movies.append(movie_data)  
</code></pre>
</blockquote>
</aside>
<p>Unless the indentation got messed up in the code block: The <code>movies.append()</code> happens <em>after</em> the for loop, not in it, so <code>movies</code> will only contain the <code>movie_data</code> for whichever result the iterator returns last. That’d explain why only one result is shown.</p>

**Comment by micaelagimenez** [comment]
<p>Oh, you are absolutely right, I put it inside and it worked! Thank you so much.</p>

**Comment by micaelagimenez** [comment]
<p>Would you know how to display the results in different boxes? Because now that I did that, I get all the results on top of each other and I am not sure how to get them in different boxes in the html. I would appreciate it if you have any ideas.</p>

**Comment by airtower-luna** [comment]
<p>Depends on what exactly you want, but in general probably wrap each result in a <code>&lt;div&gt;</code> with a CSS class that styles them as boxes. That said, I see a few issues in the HTML template that might cause issues:</p>
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="128776">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<pre><code class="lang-auto">       {% if movies %}
       {% for movie in movies %}
       &lt;div&gt;
           &lt;ul class="result"&gt;
</code></pre>
</blockquote>
</aside>
<p>Why the <code>&lt;ul&gt;</code> (“unordered list”) Element? I’d understand if you wanted to style the results as a bullet points list, but having a list without list items doesn’t make sense.</p>
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="128776">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<pre><code class="lang-auto">               &lt;style&gt;
                   @import url('https://fonts.googleapis.com/css2?family=Dosis:wght@300&amp;display=swap');
               &lt;/style&gt;
</code></pre>
</blockquote>
</aside>
<p>The style element belongs in the document header, see <a href="https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style">the MDN description</a>.</p>
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="128776">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<pre><code class="lang-auto">               &lt;h4&gt;{{movie.Title}}&lt;/h4&gt;
               &lt;h5 style="color:lightslategray;"&gt;
                   Release date: {{movie.Release_date}} &lt;br&gt;
                   Director: {{movie.Director}}
                   &lt;br&gt;Producer: {{movie.Producer}}
                   &lt;br&gt;
                   &lt;br&gt;{{movie.Description}}
               &lt;/h5&gt;
</code></pre>
</blockquote>
</aside>
<p>Having main text in a header element is questionable, and so are explicit line breaks in most cases. I think paragraphs (possibly with a class for style) or a table would work better.</p>
<aside class="quote no-group" data-username="micaelagimenez" data-post="1" data-topic="128776">
<div class="title">
<div class="quote-controls"></div>
 micaelagimenez:</div>
<blockquote>
<pre><code class="lang-auto">           &lt;/ul&gt;
           {% endfor %}

           {% endif %}
       &lt;/div&gt;
</code></pre>
</blockquote>
</aside>
<p>Each run of the template loop creates an opening <code>&lt;div&gt;</code>, but there’s only one closing <code>&lt;/div&gt;</code> at the end, so the <code>&lt;div&gt;</code> elements for the results end up nested in each other, from first to last. Browsers are pretty good at still making sense of such broken HTML, but it’s likely to cause style problems.</p>

**Comment by micaelagimenez** [comment]
<p>Oh, thank you for the observations. I changed my code a little bit but it doesn’t seem to change the overlapping results. This is what I have:</p>
<pre><code class="lang-auto">   {% if species %}
        {% for s in species %}
    &lt;div&gt;
        &lt;h4&gt;{{s.Name}}&lt;/h4&gt;
        &lt;h5 style="color:lightslategray;"&gt;
            Classification: {{s.Classification}}
            &lt;br&gt;Eye colors: {{ s.Eye_Colors }}
            &lt;br&gt;Hair colors: {{s.Hair_Colors}}
         &lt;/h5&gt;
    &lt;/div&gt;
        {% endfor %}
    {% endif %}
</code></pre>
<p>Do you know why this happens? Thank you for the help!</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>There’s very little in the HTML that might cause this. Possibly having <code>&lt;br&gt;</code> elements in a headline (<code>&lt;h5&gt;</code>), where they’re not supposed to be. But more likely it’s something caused by CSS errors, e.g. fixed height or absolute positions. Hard to say what’s really going on without seeing the active site.</p>

**Comment by micaelagimenez** [comment]
<p>Oh, it must be something in the css because since the positions for the boxes are very specific they must be piled up on top of each other. This is what happens:<img src="https://user-images.githubusercontent.com/58011910/181064945-b893804d-e6ea-474a-bd02-1f76899cbfda.png" alt="box" data-base62-sha1="ZII2BZEj3x7rDcBX4U3EIEGjQR" width="515" height="429"></p>

**Comment by micaelagimenez** [comment]
<p>I figured it out in the end! Thank you for all the help.</p>
